java/tests/com/google/privacy/differentialprivacy/BoundedSumTest.java

```diff
@@ -678,4 +680,215 @@ private static void testForBias(
+    // The result interval = (-5, 3), but it should be clamped to (0, 3), because the bounds are positive.
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(0, 3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_differentSumBoundsSigns() {
+    sum =
+            BoundedSum.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(1)
+                    .lower(-1)
+                    .upper(5)
+                    .build();
+    when(noise.computeConfidenceInterval(
+            anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,3));
+    sum.computeResult();
+
+    // The result interval = (-5, 3) and it should not be clamped.
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(-5, 3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_infiniteSumBounds() {
+    sum =
+            BoundedSum.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(1)
+                    .lower(1)
+                    .upper(5)
+                    .build();
+    when(noise.computeConfidenceInterval(
+            anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(Double.NEGATIVE_INFINITY,Double.POSITIVE_INFINITY));
+    sum.computeResult();
+
+    // The result interval = (Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY), but it should be clamped to (0, Double.POSITIVE_INFINITY).
+    assertThat(sum.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(0, Double.POSITIVE_INFINITY));
+  }
+
+  @Test
+  public void computeConfidenceInterval_gaussianTest() {
+    when(noise.computeConfidenceInterval(anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new GaussianNoise().computeConfidenceInterval(
+                    (Double) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Double) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3],  (Double) invocation.getArguments()[4],  (Double) invocation.getArguments()[5]));
+    // Mock the noise mechanism.
+    sum =
+            BoundedSum.builder()
+                    .epsilon(0.5)
+                    .delta(0.9)
+                    .noise(noise)
+                    .maxPartitionsContributed(15)
+                    .lower(1)
+                    .upper(5)
+                    .build();
+    sum.addEntry(1);
+    sum.computeResult();
+
+    // The result interval = (-0.5760855815594823, 2.5760855815594823), but it should be clamped to (0, 2.5760855815594823).
```

Comment:
Are these values mathematically exact. The want value should always be exact and computed with a different, external resource, e.g. wolfram alpha. And we should only be testing up to a certain number number of most significant digits since past that, the float value will most likely not be exact.

Considering that the sigma for the gaussian computation is only exact for the 3 most significant digits we should not be testing further than that. I had a similar conversation with Karim yesterday, you can ask him for more information :)